src/pokemon.c

```diff
@@ -0,0 +1,5567 @@
+        break;
+    }
+
+    return result;
+}
+
+s8 GetNatureStatAffinity (u8 monNature, u8 statType)
+{
+    return sNatureStatAffinities[monNature][statType - 1];
+}
+
+static const s8 Unk_020F05A0[][3] = {
+    {0x5, 0x3, 0x2},
+    {0x5, 0x3, 0x2},
+    {0x1, 0x1, 0x0},
+    {0x3, 0x2, 0x1},
+    {0x1, 0x1, 0x0},
+    {0x1, 0x1, 0x1},
+    {-1, -1, -1},
+    {-5, -5, -10},
+    {-5, -5, -10},
+    {0x3, 0x2, 0x1}
+};
+
+void sub_02075C74 (Pokemon * mon, u8 param1, u16 param2)
+{
+    u16 monSpeciesEgg;
+    u16 monHeldItem;
+    s16 monFriendship;
+    s8 v3;
+    u8 v4;
+    u8 itemHoldEffect;
+
+    if (param1 == 5) {
+        if (LCRNG_Next() & 1) {
+            return;
+        }
+    }
+
+    monSpeciesEgg = GetMonData(mon, MON_DATA_SPECIES_EGG, NULL);
+
+    if ((monSpeciesEgg == SPECIES_NONE) || (monSpeciesEgg == SPECIES_EGG)) {
+        return;
+    }
+
+    monHeldItem = GetMonData(mon, MON_DATA_HELD_ITEM, NULL);
+    itemHoldEffect = Item_LoadParam(monHeldItem, 1, 0);
+    v4 = 0;
+    monFriendship = GetMonData(mon, MON_DATA_FRIENDSHIP, NULL);
+
+    if (monFriendship >= 100) {
+        v4++;
+    }
+
+    if (monFriendship >= 200) {
+        v4++;
+    }
+
+    v3 = Unk_020F05A0[param1][v4];
+
+    if ((v3 > 0) && (GetMonData(mon, MON_DATA_POKEBALL, NULL) == 11)) {
+        v3++;
+    }
+
+    if ((v3 > 0) && (GetMonData(mon, MON_DATA_152, NULL) == param2)) {
+        v3++;
+    }
+
+    if (v3 > 0) {
+        if (itemHoldEffect == HOLD_EFFECT_FRIENDSHIP_UP) {
+            v3 = v3 * 150 / 100;
+        }
+    }
+
+    monFriendship += v3;
+
+    if (monFriendship < 0) {
+        monFriendship = 0;
+    }
+
+    if (monFriendship > 255) {
+        monFriendship = 255;
+    }
+
+    SetMonData(mon, MON_DATA_FRIENDSHIP, (u8 *)&monFriendship);
+}
+
+u8 GetMonGender (Pokemon * mon)
+{
+    return GetBoxMonGender((BoxPokemon *)&mon->box);
+}
+
+u8 GetBoxMonGender (BoxPokemon * boxMon)
+{
+    u16 monSpecies;
+    u32 monPersonality;
+    int reencrypt;
+
+    reencrypt = DecryptBoxMon(boxMon);
+    monSpecies = GetBoxMonData(boxMon, MON_DATA_SPECIES, 0);
+    monPersonality = GetBoxMonData(boxMon, MON_DATA_PERSONALITY, 0);
+
+    EncryptBoxMon(boxMon, reencrypt);
+
+    return GetMonPersonalityGender(monSpecies, monPersonality);
+}
+
+u8 GetMonPersonalityGender (u16 monSpecies, u32 monPersonality)
+{
+    PokemonPersonalData * monPersonalData;
+    u8 monGender;
+
+    monPersonalData = GetMonPersonalData(monSpecies, 0);
+    monGender = GetMonPersonalDataGender(monPersonalData, monSpecies, monPersonality);
+
+    FreeMonPersonalData(monPersonalData);
+
+    return monGender;
+}
+
+u8 GetMonPersonalDataGender (PokemonPersonalData * monPersonalData, u16 unused_monSpecies, u32 monPersonality)
+{
+    u8 monGender = GetMonPersonalDataAttribute(monPersonalData, MON_DATA_PERSONAL_GENDER);
+
+    switch (monGender) {
+    case 0:
+        return 0;
+    case 254:
+        return 1;
+    case 255:
+        return 2;
```

Comment:
From the binary community, I know that these are:

* `0` -> Always Male
* `254` -> Always Female
* `255` -> No Gender

And then, similarly, the return values are just the constant values for Male, Female, and None, respectively. Could use some consts.